components/app/explore/DatasetWidget.js

```diff
@@ -133,7 +134,7 @@ class DatasetWidget extends React.Component {
                 route={'explore_detail'}
                 params={{ id: this.props.dataset.id }}
               >
-                <a>{dataset.name}</a>
+                <a>{metadata == undefined ? dataset.name : metadata.attributes.name}</a>
```

Comment:
The metadata name that should be used is:
metadata.attributes.info.name
![image](https://user-images.githubusercontent.com/545342/30050042-3457ffd6-921d-11e7-8148-568c77f2b6b0.png)

Also:
1. please use `!metadata` instead of `metadata == undefined`
2. please always use `===` instead of `==`